modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -49,6 +51,93 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     });
   });
 
+  const toggleColumnHeader = () => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        if (isEntireColumnsHeaders(editor, selections)) {
+          editColumnsHeader(table, startCell, 'td');
+        } else {
+          editColumnsHeader(table, startCell, 'th');
+        }
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
+      });
+    });
+  };
+
+  const editColumnsHeader = (table: SugarElement<HTMLTableElement>, startCell: SugarElement<HTMLTableCellElement>, changeTo: 'th' | 'td'): void => {
+    const warehouse = Warehouse.fromTable(table);
+    const targets = TableTargets.forMenu(selections, table, startCell);
+
+    const usedColumns: number[] = [];
+    Arr.each(warehouse.all, (row) => {
```

Comment:
As mentioned elsewhere, using the warehouse to modify the table shouldn't be done here. Please create an appropriate snooker operation instead and use it via `TableActions`.

Reply:
Actually, correct me if I'm wrong but why do we even need all this logic? I believe this just tries to reimplement the `makeColumnsHeader` and `unmakeColumnsHeader` APIs.